weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx

```diff
@@ -749,15 +791,104 @@ const ExportRunsTableButton = ({
       className="mx-16"
       size="medium"
       variant="secondary"
-      onClick={() =>
-        tableRef.current?.exportDataAsCsv({includeColumnGroupsHeaders: false})
-      }
+      onClick={onClick}
       icon="export-share-upload">
-      Export to CSV
+      {selectedCalls.length > 0 ? `${selectedCalls.length}` : ''}
     </Button>
   </Box>
 );
 
+const ManageExportButton = ({
+  tableRef,
+  selectedCalls,
+  exportProps,
+}: {
+  tableRef: React.MutableRefObject<GridApiPro>;
+  selectedCalls: string[];
+  exportProps: any;
+}) => {
+  const [open, setOpen] = useState(false);
+
+  const handleExportAll = (
+    rows: Array<TraceCallSchema & {[key: string]: string}>
+  ) => {
+    // get current displayed rows
+    const currentRows = tableRef.current.getAllRowIds();
+
+    // update the table with all rows
+    tableRef.current.updateRows(rows);
+    tableRef.current.exportDataAsCsv({includeColumnGroupsHeaders: false});
+
+    const idsToDelete = rows
+      .map((row: any) => row.id)
+      .filter((id: any) => !currentRows.includes(id));
+
+    // delete the rows that added
+    tableRef.current.updateRows(
+      idsToDelete.map((rowId: any) => ({id: rowId, _action: 'delete'}))
+    );
+  };
+
+  const allPages = {pageSize: MAX_EXPORT_SIZE, page: 0};
+  const allCalls = useCallsForQuery(
+    exportProps.entity,
+    exportProps.project,
+    exportProps.effectiveFilter,
+    exportProps.filterModel,
+    exportProps.sortModel,
+    allPages,
+    exportProps.expandedRefCols
+  );
+  const [allRows, setAllRows] = useState<
```

Comment:
look at native support for csv export in the backend

Reply:
https://github.com/wandb/core/blob/ea7b072701a39f34e5e000e1fdd444bf03f9f124/services/weave-trace/src/trace_server.py#L601